### ISSUE TITLE
False positive: `unicorn/no-array-for-each` on `Effect.forEach` (Effect)

### ISSUE THREAD

**Issue by BarsukinSyn** [sindresorhus/eslint-plugin-unicorn#2749]
### Rule 
`unicorn/no-array-for-each`

### Problem 
The rule incorrectly flags `Effect.forEach` from the `Effect` ecosystem as if it were `Array.prototype.forEach`.

```ts
import { Effect } from "effect";

const program = Effect.forEach([1, 2, 3], (n) => Effect.succeed(n), { // Use `for…of` instead of `.forEach(…)`. (eslint unicorn/no-array-for-each)
  concurrency: "unbounded",
  discard: true,
});
```

The rule appears to match any `*.forEach(...)` call. In this case, `Effect.forEach` is not a method on an array value; it’s a module function.

### Rationale

A [recent fix](https://github.com/sindresorhus/eslint-plugin-unicorn/commit/0ddf5ead59925482c8a598fcf7d5f270bde15b94#diff-ad52994d9c7087aa5a04872639d1cb17f74e1731440c0dfaa59c9f7db3450ef7) for `throw-new-error` added an guard to skip Effect’s `Data.TaggedError` pattern. The same approach can be used here.

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/e5a7f54746b40355993833599736ec97936f3ab9/rules/no-array-for-each.js#L427

### Proposed fix

```js
import {isMemberExpression} from './ast/index.js';

// …
if (isMemberExpression(callee, {
  object: 'Effect',
  property: 'forEach',
  computed: false,
})) {
  return;
}
```
